.golangci.yaml

```diff
@@ -0,0 +1,82 @@
+version: "2"
+
+linters:
+  default: none
+
+  enable:
+    - errcheck
+    - ineffassign
+    - revive
+    - unused
+    - staticcheck
+    - unused
+    - unconvert
+    - misspell
+    - prealloc
+    - nakedret
+    - unparam
+
+  settings:
+    revive:
+      enable-all-rules: true
+
+      rules:
+        - name: dot-imports
+          disabled: true
+        - name: cyclomatic
+          disabled: true
+        - name: cognitive-complexity
+          disabled: true
+        - name: empty-lines
+          disabled: true
+        - name: line-length-limit
+          disabled: true
+        - name: add-constant
+          disabled: true
+        - name: package-comments
+          disabled: true
+        - name: use-errors-new
+          disabled: true
+        - name: unused-receiver
+          disabled: true
+        - name: unused-parameter
+          disabled: true
+        - name: function-length
+          disabled: true
+        - name: flag-parameter
+          disabled: true
+        - name: superfluous-else
+          disabled: true
+        - name: unexported-naming
+          disabled: true
+        - name: max-public-structs
+          disabled: true
+        - name: nested-structs
+          disabled: true
+        - name: max-control-nesting
+          disabled: true
+        - name: bare-return
+          disabled: true
+        - name: unnecessary-stmt
+          disabled: true
+        - name: confusing-results
+          disabled: true
+
+        # kevin recommends not disabling these
+        - name: unchecked-type-assertion
+          disabled: true
```

Comment:
@brianonn can you take a look at the linter results when we don't disable this?
https://github.com/PaloAltoNetworks/elemental/actions/runs/14814866494/job/41594086119